modules/drivers/presto-jdbc/src/metabase/driver/presto_jdbc.clj

```diff
@@ -0,0 +1,470 @@
+        (vary-meta assoc ::in-report-zone? true))
+      expr)))
+
+;; most date extraction and bucketing functions need to account for report timezone
+
+(defmethod sql.qp/date [:presto-jdbc :default]
+  [_ _ expr]
+  expr)
+
+(defmethod sql.qp/date [:presto-jdbc :minute]
+  [_ _ expr]
+  (hsql/call :date_trunc (hx/literal :minute) (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :minute-of-hour]
+  [_ _ expr]
+  (hsql/call :minute (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :hour]
+  [_ _ expr]
+  (hsql/call :date_trunc (hx/literal :hour) (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :hour-of-day]
+  [_ _ expr]
+  (hsql/call :hour (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :day]
+  [_ _ expr]
+  (hsql/call :date_trunc (hx/literal :day) (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :day-of-week]
+  [_ _ expr]
+  (sql.qp/adjust-day-of-week :presto-jdbc (hsql/call :day_of_week (in-report-zone expr))))
+
+(defmethod sql.qp/date [:presto-jdbc :day-of-month]
+  [_ _ expr]
+  (hsql/call :day (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :day-of-year]
+  [_ _ expr]
+  (hsql/call :day_of_year (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :week]
+  [_ _ expr]
+  (sql.qp/adjust-start-of-week :presto-jdbc (partial hsql/call :date_trunc (hx/literal :week)) (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :month]
+  [_ _ expr]
+  (hsql/call :date_trunc (hx/literal :month) (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :month-of-year]
+  [_ _ expr]
+  (hsql/call :month (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :quarter]
+  [_ _ expr]
+  (hsql/call :date_trunc (hx/literal :quarter) (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :quarter-of-year]
+  [_ _ expr]
+  (hsql/call :quarter (in-report-zone expr)))
+
+(defmethod sql.qp/date [:presto-jdbc :year]
+  [_ _ expr]
+  (hsql/call :date_trunc (hx/literal :year) (in-report-zone expr)))
+
+(defmethod sql.qp/unix-timestamp->honeysql [:presto-jdbc :seconds]
+  [_ _ expr]
+  ;; ensure that unix timestamps are always interpreted in UTC regardless of report zone
+  (hsql/call :from_unixtime expr (hx/literal "UTC")))
+
+(defmethod sql.qp/unix-timestamp->honeysql [:presto-jdbc :milliseconds]
+  [_ _ expr]
+  ;; from_unixtime doesn't support milliseconds directly, but we can add them back in
+  (let [millis (hsql/call (u/qualified-name ::mod) expr 1000)]
+    (hsql/call :date_add
+               (hx/literal "millisecond")
+               millis
+               (hsql/call :from_unixtime (hsql/call :/ expr 1000) (hx/literal "UTC")))))
+
+(defmethod sql.qp/unix-timestamp->honeysql [:presto-jdbc :microseconds]
+  [_ _ expr]
+  ;; Presto can't even represent microseconds, so convert to millis and call that version
+  (sql.qp/unix-timestamp->honeysql :presto-jdbc :milliseconds (hsql/call :/ expr 1000)))
```

Comment:
Same thing applies here, you should pass `driver` rather than passing `:presto-jdbc` literally. You never know if we might want to use `:presto-jdbc` as the parent of something else in the future. (e.g. maybe in a test we want to mock out some situations so we create `::presto-jdbc-test`). It doesn't matter now since we have no child drivers, but it's better practice to write the code so it doesn't break if we add them